datafusion/src/optimizer/simplify_expressions.rs

```diff
@@ -1262,25 +1252,8 @@ mod tests {
         // Make sure c1 column to be used in tests is not boolean type
         assert_eq!(col("c1").get_type(&schema).unwrap(), DataType::Utf8);
 
-        // don't fold c1 = true
-        assert_eq!(
```

Comment:
as c1 is a string column, it is a runtime error to try and evaluate `c1 = true` -- I fixed the case and removed the nonsensical tests